node_modules/col-assets/lib/api.js

```diff
@@ -513,7 +517,25 @@ var getAssets = module.exports.getAssets = function(ctx, filters, sort, limit, o
           });
         });
 
-        return callback(null, data);
+        var options = {
+          'where': {
+            'asset_id': assetIds
+          }
+        };
+        DB.Pin.findAll(options).complete(function(err, pins) {
```

Comment:
@paulkerschen Okay, no luck here. I've tested with permutations similar to suggestion 
 in sequelize PR 1830 above. Few observations:
* This block has a similar `DB.User.findAll` nested under the raw SQL. Do you think it's also unnecessary?
* I notice that other definitions for `Db.Asset` model (e.g., Category) do not influence `getAssets` results.  Whereas, those definitions do matter in the more textbook `getAsset` case.
* I just removed the following from raw SQL in question. (pin_count is not needed, after all.) Thus a small performance improvement.
```
(select count(*) from pinned_user_assets p where p.asset_id = a.id)::int AS pin_count
```

I suggest we create a new Jira to measure and improve performance of `getAssets`. In the process, we might eliminate the nested `DB.User.findAll`. Thoughts?